MapboxVision/VideoStream/VisionManager.swift

```diff
@@ -710,6 +696,35 @@ public final class VisionManager {
     private var currentFrame: CVPixelBuffer?
 }
 
+extension VisionManager: VideoSourceObserver {
+    public func videoSource(_ videoSource: VideoSource, didOutput videoSample: VideoSample) {
+        guard let pixelBuffer = videoSample.buffer.pixelBuffer else {
+            assertionFailure("Sample buffer containing pixel buffer is expected here")
+            return
+        }
+        
+        presenter?.present(sampleBuffer: videoSample.buffer)
+        
+        currentFrame = pixelBuffer
+        
+        guard state.isStarted else { return }
+        
+        dependencies.recorder.handleFrame(videoSample.buffer)
+        
+        dependencies.core.setImage(pixelBuffer)
+    }
+    
+    public func videoSource(_ videoSource: VideoSource, didOutput cameraParameters: CameraParameters) {
+        // TODO: use new camera params
+        dependencies.core.setCameraWidth(
```

Comment:
Is method named `setCameraWidth`?
Also, are `-1` values really intended here?

Reply:
There's no fields for focal length in pixels in current core API. It'll change in a new version and the correct method will be utilized here.